framework-api-core/src/main/java/uk/gov/justice/services/core/enveloper/Enveloper.java

```diff
@@ -24,4 +27,43 @@
      * @return a function that wraps objects into an envelope.
      */
     Function<Object, JsonEnvelope> withMetadataFrom(final JsonEnvelope envelope, final String name);
+
+    static Function<Object, JsonEnvelope> toEnvelopeWithMetadataFrom(final Envelope<?> envelope) {
+        return EnveloperProvider.provider().toEnvelopeWithMetadataFrom(envelope);
+    }
+
+    static <T> EnveloperBuilder<T> envelop(final T payload) {
+        return EnveloperProvider.provider().envelop(payload);
+    }
+
+    /**
+     * A helper interface to provide an instance of a {@link Envelope} with given {@link Metadata}
+     */
+    interface EnveloperBuilder<T> {
+
+        /**
+         * Helper interface method to set name
+         *
+         * @param name the name to be added to the Envelope.
+         * @return the EnveloperBuilder instance
+         */
+        EnveloperBuilder<T> withName(String name);
+
+        /**
+         * Helper interface method to apply metadata
+         *
+         * @param envelope the Envelope with metadata to be applied on new envelope.
+         * @return the EnveloperBuilder instance
+         */
+        Envelope<T> withMetadataFrom(final Envelope<?> envelope);
+
+        /**
+         * Helper interface method to apply metadata
+         *
+         * @param jsonEnvelope the JsonEnvelopewith metadata to be applied on new envelope
+         * @return the EnveloperBuilder instance
+         */
+        JsonEnvelope withMetadataFrom(final JsonEnvelope jsonEnvelope);
```

Comment:
I think we should ditch this and force people to use Envelope<JsonValue> if they want to use the new interface